api/controlplane/kubeadm/v1beta2/kubeadm_control_plane_types.go

```diff
@@ -540,31 +526,60 @@ type KubeadmControlPlaneMachineTemplateDeletionSpec struct {
 	NodeDeletionTimeoutSeconds *int32 `json:"nodeDeletionTimeoutSeconds,omitempty"`
 }
 
-// RolloutBefore describes when a rollout should be performed on the KCP machines.
-type RolloutBefore struct {
+// KubeadmControlPlaneRolloutSpec allows you to configure the behaviour of rolling updates to the control plane Machines.
+// It allows you to require that all Machines are replaced before or after a certain time,
+// and allows you to define the strategy used during rolling replacements.
+// +kubebuilder:validation:MinProperties=1
+type KubeadmControlPlaneRolloutSpec struct {
+	// before is a field to indicate a rollout should be performed
+	// if the specified criteria is met.
+	// +optional
+	Before KubeadmControlPlaneRolloutBeforeSpec `json:"before,omitempty,omitzero"`
+
+	// after is a field to indicate a rollout should be performed
+	// after the specified time even if no changes have been made to the
+	// KubeadmControlPlane.
+	// Example: In the YAML the time can be specified in the RFC3339 format.
+	// To specify the rolloutAfter target as March 9, 2023, at 9 am UTC
+	// use "2023-03-09T09:00:00Z".
+	// +optional
+	After *metav1.Time `json:"after,omitempty"`
```

Comment:
`metav1.Time` now exposes [IsZero](https://github.com/kubernetes/apimachinery/blob/58c4eb072ebfc89d9219c4dbc72e885612cff52f/pkg/apis/meta/v1/time.go#L61-L66) which means that you could switch this around
```suggestion
	After metav1.Time `json:"after,omitempty,omitzero"`
```

I assume at least that we don't allow setting the zero value as a valid choice here

Reply:
I have it working, but I'll have to make this a separate PR, this requires changes in our SSA helper because the v0.33  unstructured converter does not implement omitzero yet (v0.34 does)